### PR TITLE
add logic of ending quiz

### LINKED ISSUE #8: Implement logic to detect end of the quiz
When the user answers the last question, the app needs to show the summary page.

### PR DESCRIPTION
**Pull request by belal19979**
closes #8

### CODE REVIEW

File: src/views/detectEndOfQuiz.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { quizData } from "../data.js";
+import { finalSummaryPage } from "../pages/finalSummaryPage.js";
+
+export const detectEndOfQuiz = () => {
```
Comment (arkaitzgarro) [inline src/views/detectEndOfQuiz.js:4]: Where do you plan to call this question? And technically speaking, this is not a view, because it doesn't render anything :)

Comment (belal19979) [reply src/views/detectEndOfQuiz.js:4]: should I put it in pages? I thought pages for rendering page :)
I was supposed to put it in the file where will gather all functions together,, I think app.js may be.

Comment (arkaitzgarro) [reply src/views/detectEndOfQuiz.js:4]: This function needs to be executed every time you answer a question right? Maybe it can give you a hint on where to place it ;)

Comment (belal19979) [reply src/views/detectEndOfQuiz.js]: I think the logic of every time answering a question is not ready yet.

Comment (arkaitzgarro) [reply src/views/detectEndOfQuiz.js]: There's something in place already :) 

https://github.com/arkaitzgarro/browser-quiz-project-starter/blob/master/src/pages/questionPage.js#L34

Comment (arkaitzgarro) [reply src/views/detectEndOfQuiz.js]: Although this won't work because we don't show the button in the last question. Forget about it.

File: src/pages/questionPage.js [hunk 2]
```diff
@@ -8,13 +8,17 @@ import {
...
+import { detectEndOfQuiz } from './detectEndOfQuiz'
+
 
 export const initQuestionPage = () => {
   const userInterface = document.getElementById(USER_INTERFACE_ID);
   userInterface.innerHTML = '';
 
   const currentQuestion = quizData.questions[quizData.currentQuestionIndex];
 
+  detectEndOfQuiz()
```
Comment (arkaitzgarro) [inline src/pages/questionPage.js:20]: This may render the summary page, but since you are not stopping the execution, it will continue with the code bellow and try to render the question as well.

Did you try it in the browser?

Comment (belal19979) [reply src/pages/questionPage.js:20]: actullay I wanted to put it inside if condition like this 
`if (currentQuestionIndex === totalQuestions - 1) {
detectEndOfQuiz()
}else{
rest of the code 
}`
but that mean i need to remove the if condition inside the function .
my opinion function should excute every time we click on answer but I do not see this function yet

Comment (arkaitzgarro) [reply src/pages/questionPage.js:20]: Then it means we have a dependency with the answer implementation, it makes sense for the condition to be part of the answer question https://github.com/arkaitzgarro/browser-quiz-project-starter/pull/15

I would suggest to talk to @AsiyeGokalp and help her finishing the task.

Comment (belal19979) [reply src/pages/questionPage.js:20]: ok Thanks

**Comment by arkaitzgarro** [comment]
Hi @belal19979, what's the state of this task? Is it ready to be reviewed?

File: src/app.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 import { quizData } from './data.js';
 import { initWelcomePage } from './pages/welcomePage.js';
-import { finalSummaryPage } from './pages/finalSummaryPage.js';
+// import { finalSummaryPage } from './pages/finalSummaryPage.js';
```
Comment (arkaitzgarro) [inline src/app.js:5]: This can be removed.

File: src/pages/questionPage.js [hunk 4]
```diff
@@ -25,11 +26,13 @@ export const initQuestionPage = () => {
     const answerElement = createAnswerElement(key, answerText);
     answersListElement.appendChild(answerElement);
   }
-
-
+  document.getElementById(NEXT_QUESTION_BUTTON_ID).classList.remove('hide');
```
Comment (arkaitzgarro) [inline src/pages/questionPage.js:29]: Why is this here?

File: src/pages/questionPage.js [hunk 5]
```diff
@@ -46,12 +49,15 @@ export const initQuestionPage = () => {
     } else {
       selectedAnswer.classList.add('wrong');
     }
+
+    if (quizData.currentQuestionIndex === quizData.questions.length - 1) {
```
Comment (arkaitzgarro) [inline src/pages/questionPage.js:53]: This needs to happen at the very end of the function, otherwise we won't save the answer for the last question.

File: sssssssssssssssssssssssssssssssss [hunk 6]
```diff
@@ -0,0 +1,254 @@
+[33mcommit d349db4eafd5b3cdc964992c7521a385f0096d46[m[33m ([m[1;36mHEAD -> [m[1;32mImplement-logic-to-detect-end-of-the-quiz[m[33m)[m
```
Comment (arkaitzgarro) [inline sssssssssssssssssssssssssssssssss:1]: 👀

Comment (belal19979) [reply sssssssssssssssssssssssssssssssss:1]: I did not make it

File: src/pages/questionPage.js (hunk 4 above)
Comment (belal19979) [reply src/pages/questionPage.js:29]: it is asyies code

Comment (arkaitzgarro) [reply src/pages/questionPage.js:29]: I don't think so, because it shows as an addition. Maybe the merge went wrong.

But anyway, please delete it.

File: sssssssssssssssssssssssssssssssss (hunk 6 above)
Comment (arkaitzgarro) [reply sssssssssssssssssssssssssssssssss:1]: Please remove the file 🔪

**Review by arkaitzgarro — APPROVED** [review]
